src/term.js

```diff
@@ -1225,13 +1225,13 @@ Terminal.prototype.refresh = function(start, end) {
 
     attr = this.defAttr;
     i = 0;
-
-    for (; i < width; i++) {
+    var widthd = width;
+    for (; i < widthd; i++) {
       data = line[i][0];
       ch = line[i][1];
 
       if (i === x) data = -1;
-
+      if (ch.charCodeAt()>=127){widthd=widthd-0.7}
```

Comment:
I'm pretty sure this should not be every character over 127 - there are many characters beyond that which should still be displayed as single width, like `£` (163), or `ø` (248). Those are just examples, there are many, many more, and working out which ones need width adjustment is tricky.

I'm also not sure where the 0.7 comes from. This may work for the wide characters of one particular font, but I wouldn't rely on it working in general.

There [is already a mechanism](https://github.com/chjj/term.js/blob/v0.0.7/src/term.js#L5878) for dealing with wide characters, which works by allowing them twice the normal space in the grid. However, it doesn't know about all the wide characters, as described in #66.